actors/ppo.py

```diff
@@ -0,0 +1,720 @@
+        self.output = torch.nn.Linear(num_hidden, 1)
+
+    def forward(self, x: torch.Tensor):
+        # Input layer
+        x = self.input(x)
+        x = torch.tanh(x)
+
+        # Hidden layer
+        x = self.fully_connected(x)
+        x = torch.tanh(x)
+
+        # Output layer
+        value = self.output(x)
+
+        return value
+
+
+@dataclass
+class NormalizationParams:
+    """Normalization paramters for state and rewards"""
+
+    mean_state: torch.Tensor  # Statistical mean for states
+    var_state: torch.Tensor  # Statistical variance for states
+    mean_reward: torch.Tensor  # Statistical mean for reward
+    var_reward: torch.Tensor  # Statistical variance for reward
+
+
+def initialize_weight(param) -> None:
+    """Orthogonal initialization of the weight's values of a network"""
+
+    if isinstance(param, torch.nn.Linear):
+        torch.nn.init.orthogonal_(param.weight.data)
+        torch.nn.init.constant_(param.bias.data, 0)
+
+
+class Normalization:
+    """Normalize the states and rewards on the fly
+    Calulates the running mean and std of a data stream
+    https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm
+    Source: https://github.com/DLR-RM/stable-baselines3
+    """
+
+    def __init__(self, dtype: torch.FloatTensor = torch.float, epsilon: float = 1e-4, nums: int = 1):
+        self.mean = torch.zeros(1, nums, dtype=dtype)
+        self.var = torch.ones(1, nums, dtype=dtype)
+        self.count = epsilon
+
+    def update(self, x: torch.Tensor):
+        batch_mean = torch.mean(x, axis=0)
+        batch_var = torch.var(x, axis=0)
+        batch_count = x.size(dim=0)
+        self.update_mean_var(batch_mean, batch_var, batch_count)
+
+    def update_mean_var(self, batch_mean: torch.Tensor, batch_var: torch.Tensor, batch_count: int):
+        self.mean, self.var, self.count = self.update_mean_var_count(
+            self.mean, self.var, self.count, batch_mean, batch_var, batch_count
+        )
+
+    @staticmethod
+    def update_mean_var_count(
+        mean: torch.Tensor,
+        var: torch.Tensor,
+        count: int,
+        batch_mean: torch.Tensor,
+        batch_var: torch.Tensor,
+        batch_count: int,
+    ) -> Tuple[torch.Tensor, int]:
+
+        """Update the statistical mean and variance"""
+        delta = batch_mean - mean
+        tot_count = count + batch_count
+        new_mean = mean + delta * batch_count / tot_count
+        mean_a = var * count
+        mean_b = batch_var * batch_count
+        mean_2 = mean_a + mean_b + (delta**0.5) * count * batch_count / tot_count
+        new_var = mean_2 / tot_count
+        new_count = tot_count
+
+        return new_mean, new_var, new_count
+
+
+class PPOModel(Model):
+    """Proximal Policy Optimization (PPO) is an on-policy algorithm.
+    https://arxiv.org/pdf/1707.06347.pdf.
+
+    Attributes:
+        model_id: Model identity
+        environment_implementation: Environment specs (not sure what it is)
```

Comment:
not sure what it is? :D

Reply:
I have replaced with more descriptive comments :)